pkg/relayer/proxy/http_server.go

```diff
@@ -141,40 +147,63 @@ func (server *relayMinerHTTPServer) Stop(ctx context.Context) error {
 // Ping tries to dial the suppliers backend URLs to test the connection.
 func (server *relayMinerHTTPServer) Ping(ctx context.Context) error {
 	for _, supplierCfg := range server.serverConfig.SupplierConfigsMap {
-		c := &http.Client{Timeout: 2 * time.Second}
-
-		backendUrl := *supplierCfg.ServiceConfig.BackendUrl
-		if backendUrl.Scheme == "ws" || backendUrl.Scheme == "wss" {
-			// TODO_IMPROVE: Consider testing websocket connectivity by establishing
-			// a websocket connection instead of using an HTTP connection.
-			server.logger.Warn().Msgf(
-				"backend URL %s scheme is a %s, switching to http to check connectivity",
-				backendUrl.String(),
-				backendUrl.Scheme,
-			)
-
-			if backendUrl.Scheme == "ws" {
-				backendUrl.Scheme = "http"
-			} else {
-				backendUrl.Scheme = "https"
-			}
+		// Initialize the backend URLs to test with the default service config.
+		backendUrls := []*url.URL{
+			supplierCfg.ServiceConfig.BackendUrl,
+		}
+
+		// Add the RPC type specific service configs to the backend URLs to test, if any.
+		for _, rpcTypeBackendURL := range supplierCfg.RPCTypeServiceConfigs {
+			backendUrls = append(backendUrls, rpcTypeBackendURL.BackendUrl)
 		}
-		resp, err := c.Head(backendUrl.String())
-		if err != nil {
-			return fmt.Errorf(
-				"failed to ping backend %q for serviceId %q: %w",
-				backendUrl.String(), supplierCfg.ServiceId, err,
-			)
+
+		// Test the connectivity of all the backend URLs for the supplier.
+		for _, backendUrl := range backendUrls {
+			if err := server.testBackendURL(backendUrl, supplierCfg.ServiceId); err != nil {
+				return err
+			}
 		}
-		_ = resp.Body.Close()
 
-		if resp.StatusCode >= http.StatusInternalServerError {
-			return fmt.Errorf(
-				"failed to ping backend %q for serviceId %q: received status code %d",
-				backendUrl.String(), supplierCfg.ServiceId, resp.StatusCode,
-			)
+	}
+
+	return nil
+}
+
+// testBackendURL tests the connectivity of a backend URL for a given service ID.
+func (server *relayMinerHTTPServer) testBackendURL(backendUrl *url.URL, serviceId string) error {
+	// Initialize the HTTP client for the backend URL.
+	c := &http.Client{Timeout: 2 * time.Second}
+
+	if backendUrl.Scheme == "ws" || backendUrl.Scheme == "wss" {
+		// TODO_IMPROVE: Consider testing websocket connectivity by establishing
+		// a websocket connection instead of using an HTTP connection.
+		server.logger.Warn().Msgf(
+			"backend URL %s scheme is a %s, switching to http to check connectivity",
+			backendUrl.String(),
+			backendUrl.Scheme,
+		)
+
+		if backendUrl.Scheme == "ws" {
+			backendUrl.Scheme = "http"
+		} else {
+			backendUrl.Scheme = "https"
 		}
+	}
```

Comment:
1. Can this be moved into a helper?
2. Could/should this logic be reused elsewhere in the codebase?

Reply:
1. Done
2. Not as far as I know; I don't believe we use this anywhere else. A search of the codebase for `== "wss"` & `== "ws"` confirms this.